README.md

```diff
@@ -1,49 +1,86 @@
 # JSON Schema usage and problems
 
-Public forum to collect real world schemas and to discuss JSON Schema specification problems
-
 [![Build Status](https://travis-ci.org/epoberezkin/json-schema-usage-problems.svg?branch=master)](https://travis-ci.org/epoberezkin/json-schema-usage-problems)
 
+This repo is a public forum for collecting real-world schemas and discussing
+problems with the JSON Schema specification.
 
 ## Problems
 
-For JSON Schema authors and users, particularly those who are just starting to use JSON Schema, the specification is quite confusing. Validators, while complying with the spec, often produce the results that surprise users - ignored unknown keywords, type is not enforced, properties are optional, etc.
-
-For validator maintainers, the current draft-07 of the JSON Schema specification is already quite difficult to implement (See [Test of compliance of JS validators with the specification][1]). The upcoming draft-08 includes additional features that further increase implementation complexity.
-
-For tool maintainers, JSON Schema presents further challenges that have been partially addressed in OpenAPI version of the specification, but having two different versions of JSON Schema also creates additional problems for the users.
-
+At a high level, JSON Schema has three big problems:
+
+* **Bad default behavior.** For JSON Schema authors and users, particularly
+  those new to JSON Schema, the spec is very confusing and surprising. Some
+  particularly bad defaults are:
+
+  1. Keywords that are specific to particular data types, like `items` or
```

Comment:
"bad" is too generic. Break it down.

Broadly speaking, the issues that you chose as examples identify a fail-silent system.

> **Fail-silent system.** JSON Schema ignores keywords that are specific to particular data types (like `items` or `properties`) if the inputted data isn't of the relevant data type. In practise, this means that unknowing the schema owner, the schema is not enforcing the expected constraints (e.g. in case author made a typo in the spec). Example:
> [give example]

The rest could be described as "Disconnected hierarchy", e.g.

> **Disconnected hierarchy.** JSON Schema defines multiple keywords that affect behaviour of other keywords without owning them, e.g. `properties` keyword defines object properties, but a separate keyword `required` lists which properties are required. In practise, this means that you must reference and be aware of references to a keyword in order to understand the behaviour of a JSON Schema fragment.